terraform/Makefile

```diff
@@ -0,0 +1,13 @@
+all: plan
+
+fmt:
+	terraform fmt
+
+plan: fmt
+	terraform plan -out proposed.plan
+
+apply:
+	terraform apply proposed.plan
```

Comment:
Plan looks reasonable, but I don't want to apply this anywhere near customer data etc.

@malcolmsparks do you have somewhere I can test this out tomorrow? If I can get an example user data script that configures things and starts up the services I think this work will be more or less done.

```
terraform fmt
terraform plan -out proposed.plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.template_file.user_data: Refreshing state...
data.aws_ami.rock: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_instance.rock
      id:                                        <computed>
      ami:                                       "ami-cec09ab7"
      associate_public_ip_address:               <computed>
      availability_zone:                         "eu-west-1"
      ebs_block_device.#:                        <computed>
      ephemeral_block_device.#:                  <computed>
      get_password_data:                         "false"
      instance_state:                            <computed>
      instance_type:                             "t2.small"
      ipv6_address_count:                        <computed>
      ipv6_addresses.#:                          <computed>
      key_name:                                  "rock"
      network_interface.#:                       <computed>
      network_interface_id:                      <computed>
      password_data:                             <computed>
      placement_group:                           <computed>
      primary_network_interface_id:              <computed>
      private_dns:                               <computed>
      private_ip:                                <computed>
      public_dns:                                <computed>
      public_ip:                                 <computed>
      root_block_device.#:                       "1"
      root_block_device.0.delete_on_termination: "true"
      root_block_device.0.volume_id:             <computed>
      root_block_device.0.volume_size:           "20"
      root_block_device.0.volume_type:           <computed>
      security_groups.#:                         <computed>
      source_dest_check:                         "true"
      subnet_id:                                 <computed>
      tags.%:                                    "2"
      tags.Name:                                 "rock"
      tags.Source:                               "juxt/rock"
      tenancy:                                   <computed>
      user_data:                                 "8aed11c63960f4590c9cdd426593ecba98c4ab69"
      volume_tags.%:                             <computed>
      vpc_security_group_ids.#:                  <computed>

  + aws_security_group.allow_all
      id:                                        <computed>
      arn:                                       <computed>
      description:                               "Allow all inbound traffic."
      egress.#:                                  <computed>
      ingress.#:                                 "1"
      ingress.482069346.cidr_blocks.#:           "1"
      ingress.482069346.cidr_blocks.0:           "0.0.0.0/0"
      ingress.482069346.description:             ""
      ingress.482069346.from_port:               "0"
      ingress.482069346.ipv6_cidr_blocks.#:      "0"
      ingress.482069346.protocol:                "-1"
      ingress.482069346.security_groups.#:       "0"
      ingress.482069346.self:                    "false"
      ingress.482069346.to_port:                 "0"
      name:                                      "allow_all"
      owner_id:                                  <computed>
      revoke_rules_on_delete:                    "false"
      tags.%:                                    "2"
      tags.Name:                                 "allow_all"
      tags.Source:                               "juxt/rock"
      vpc_id:                                    <computed>


Plan: 2 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

This plan was saved to: proposed.plan

To perform exactly these actions, run the following command to apply:
    terraform apply "proposed.plan"
```